### PR TITLE
Remove BaggageUtils

### PR DESCRIPTION
**Pull request by anuraaga**
Since it's private, it's relatively harmless to have, but does cause the JVM optimizer a bit more work to inline more methods and somewhat abstracts the fact that BaggageUtils is really just a holder of a context key so seems nice to remove. Another option is to rename to `BaggageContextKey` with the lone field inside, and still removing the methods. Or just not do anything is fine too but lean towards removing this.

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
If for some reason we feel like this is necessary in the future, it would be easy enough to add it back. Removing unneeded code is a win, for sure. Thanks!

**Comment by carlosalberto** [comment]
Although this change is OK, it feels _very_ weird to have the key in a class called `ImmutableBaggage`. Isn't this key used for **all** `Baggage` instances?

File: api/src/test/java/io/opentelemetry/api/baggage/BaggageContextTest.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ void testGetCurrentBaggage_Default() {
   @Test
   void testGetCurrentBaggage_SetCorrContext() {
```
Comment (jkwatson) [inline api/src/test/java/io/opentelemetry/api/baggage/BaggageContextTest.java:25]: this test method name is pretty out of date..still referring to it as "correlationContext"

**Comment by jkwatson** [comment]
> Although this change is OK, it feels _very_ weird to have the key in a class called `ImmutableBaggage`. Isn't this key used for **all** `Baggage` instances?

I'm sympathetic to this point of view. Maybe we want a simple package-private "BaggageKey" class that holds on to the key?

**Pull request merged**